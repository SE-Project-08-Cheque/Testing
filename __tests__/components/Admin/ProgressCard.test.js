// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import ProgressCard from "../../../components/Admin/ProgressCard";


afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ProgressCard />, div);
});

it("renders progress card correctly", () => {
  const { getByTestId } = render(
    <ProgressCard
     currentvalue = {25}
     Heading = "Asisstants"

    />
  );
  expect(getByTestId('current_value')).toHaveTextContent("25");
  expect(getByTestId('heading')).toHaveTextContent("Asisstants");
});

it("matches snapshot", () => {
  const tree = renderer.create(<ProgressCard
    currentvalue = {25}
    heading = "Asisstants"

   />);

    expect(tree).toMatchSnapshot();
})

