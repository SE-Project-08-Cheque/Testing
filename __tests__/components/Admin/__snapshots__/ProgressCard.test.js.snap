// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-khhas1"
>
  <p
    className="chakra-text css-gnndzk"
    data-testid="current_value"
  >
    25
  </p>
  <p
    className="chakra-text css-e1ztz8"
    data-testid="heading"
  />
</div>
`;
